### ISSUE TITLE
Git(Hub) and remote connection controls

### ISSUE THREAD

**Issue by hanaral** [elementary/code#941]
## Problem

It can't be expected that developers have to bounce from once app to another to do something so integral to elementary app development.

## Proposal

Add buttons at the top of te sidebar to commit, add comments etc, and add a way through System Settings > Online Accounts to sign in to GitHub or another git service. 

## Prior Art

Panic Nova's settings menu | Nova's git control sidebar
------------ | -------------
![Screenshot from 2021-02-06 00-56-12](https://user-images.githubusercontent.com/58219504/107116174-62994a00-6869-11eb-8b88-75d182dab84b.png) | ![editor-1](https://user-images.githubusercontent.com/58219504/107116234-a8eea900-6869-11eb-8140-5bfa8db56790.png)

**Comment by kaixoo** [comment]
I agree, but a generic Git interface is more important. Signing in through Online Accounts and automatically setting the git user/email would be great indeed.

Maybe implement as a plugin at first then look into integrating into Code?

**Comment by jeremypw** [comment]
I hope to add some simple push, pull and commit functionality at some point but including a full-blown Git gui is unlikely - there are several good ones already in the App Store and it is not a great burden to `Alt-Tab` between two windows or swipe between two workspaces.
